src/packages/objectmanager/objectmanager.js

```diff
@@ -41,21 +43,19 @@ export class ObjectManager {
      * @return {void}
      */
     removeObject({name}) {
-        delete this.objects[name]
+        this.objects.delete(name)
     }
 
     /**
      * Reorders the objects based on their z index
      * @return {void}
      */
     reorderObjects() {
-        let orderedObjects = Object.values(this.objects)
-        orderedObjects.sort((function (a, b) {
-            if (a.z < b.z) return -1;
-            if (a.z > b.z) return 1;
-            return this.indexOf(a) - this.indexOf(b);
-        }).bind(orderedObjects))
-        return orderedObjects
+        this.objects = new Map([...this.objects].sort(function (a, b) {
+            if (a[1].z < b[1].z) return -1;
+            if (a[1].z > b[1].z) return 1;
+            return 0
+        }))
```

Comment:
Kannst du hier nicht einfach ``return a[1].z - b[1].z`` machen? Oder müssen das genau 1, -1 und 0 sein?

Reply:
Ja klar, kann man auch machen. Wollte sicher gehen, dass höhere Zahlen keinen Einfluss auf die Gewichtung der Verschiebung in der Sortierung nehmen.